modules/account-lib/src/coin/sol/solInstructionFactory.ts

```diff
@@ -98,16 +99,40 @@ function memoInstruction(data: Memo): TransactionInstruction[] {
  */
 function transferInstruction(data: Transfer): TransactionInstruction[] {
   const {
-    params: { fromAddress, toAddress, amount },
+    params: { fromAddress, toAddress, amount, mint, owner },
   } = data;
   assert(fromAddress, 'Missing fromAddress param');
   assert(toAddress, 'Missing toAddress param');
   assert(amount, 'Missing toAddress param');
-  const transferInstruction = SystemProgram.transfer({
-    fromPubkey: new PublicKey(fromAddress),
-    toPubkey: new PublicKey(toAddress),
-    lamports: new BigNumber(amount).toNumber(),
-  });
+  let transferInstruction;
+  if (!mint) {
+    // native transfer
+    transferInstruction = SystemProgram.transfer({
+      fromPubkey: new PublicKey(fromAddress),
+      toPubkey: new PublicKey(toAddress),
+      lamports: new BigNumber(amount).toNumber(),
+    });
+  } else {
+    // token transfer
+    assert(mint, 'Missing mint');
+    assert(owner, 'Missing owner');
+    let decimalPlaces;
+    coins.forEach((value, key) => {
+      if (value['tokenAddress'] === mint) {
```

Comment:
add a check here for whether the coin is solana, on the unlikely event that the same mint address is valid on multiple chains